src/pt/sussyscan/src/eu/kanade/tachiyomi/extension/pt/sussyscan/SussyScan.kt

```diff
@@ -1,34 +1,220 @@
+        .set("scan-id", "1") // Required header for requests
+
+    // ============================= Popular ==================================
 
-    override val mangaDetailsSelectorTitle = "${super.mangaDetailsSelectorTitle}, span.rate-title, title"
-    override val mangaDetailsSelectorThumbnail = "head meta[property='og:image']"
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$apiUrl/obras/top5", headers)
+    }
 
-    override fun mangaDetailsParse(document: Document) = super.mangaDetailsParse(document).apply {
-        title = title.substringBeforeLast("–")
+    override fun popularMangaParse(response: Response): MangasPage {
+        val dto = response.parseAs<WrapperDto<List<MangaDto>>>()
+        val mangas = dto.results.map { it.toSManga() }
+        return MangasPage(mangas, false) // There's a pagination bug
     }
 
-    override fun imageFromElement(element: Element): String? {
-        return super.imageFromElement(element)?.takeIf { it.isNotEmpty() }
-            ?: element.attr("content") // Thumbnail from <head>
+    // ============================= Latest ===================================
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$apiUrl/obras/novos-capitulos".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", page.toString())
+            .addQueryParameter("limite", "24")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage {
+        val dto = response.parseAs<WrapperDto<List<MangaDto>>>()
+        val mangas = dto.results.map { it.toSManga() }
+        return MangasPage(mangas, dto.hasNextPage())
+    }
+
+    // ============================= Search ===================================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/obras".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", page.toString())
+            .addQueryParameter("limite", "8")
+            .addQueryParameter("obr_nome", query)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response) = latestUpdatesParse(response)
+
+    // ============================= Details ==================================
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val url = "$apiUrl/obras".toHttpUrl().newBuilder()
+            .addPathSegment(manga.id)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun mangaDetailsParse(response: Response) =
+        response.parseAs<WrapperDto<MangaDto>>().results.toSManga()
+
+    private val SManga.id: String get() {
+        val url = apiUrl.toHttpUrl().newBuilder()
+            .addPathSegments(url)
+            .build()
+        return url.pathSegments[2]
+    }
```

Comment:
Local variable `url` is shadowing `url` from `SManga`, which may cause confusion.